osu.Game/Graphics/Containers/StoryboardContainer.cs

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Allocation;
+using osu.Framework.Graphics;
+using osu.Game.Storyboards;
+using osu.Game.Storyboards.Drawables;
+
+namespace osu.Game.Graphics.Containers
+{
+    /// <summary>
+    /// A container that handles <see cref="Storyboard"/> loading, as well as applies user-specified visual settings to it.
+    /// </summary>
+    public class StoryboardContainer : UserDimContainer
+    {
+        private readonly Storyboard storyboard;
+        private DrawableStoryboard drawableStoryboard;
+
+        public StoryboardContainer(Storyboard storyboard)
+        {
+            this.storyboard = storyboard;
+        }
+
+        [BackgroundDependencyLoader]
+        private void load()
+        {
+            initializeStoryboard(false);
+        }
+
+        protected override void LoadComplete()
+        {
+            ShowStoryboard.ValueChanged += _ => initializeStoryboard(true);
```

Comment:
why not just use `BindValueChanged(...,true)` and fix the other issue you opened?

there's no point in transferring bugs in code you're basically rewriting.

Reply:
was going to make a separate PR in case you wanted to further scrutinize it, but I guess its a simple enough change.